cmd/helm/uninstall.go

```diff
@@ -64,7 +65,7 @@ func newUninstallCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 	f := cmd.Flags()
 	f.BoolVar(&client.DryRun, "dry-run", false, "simulate a uninstall")
 	f.BoolVar(&client.DisableHooks, "no-hooks", false, "prevent hooks from running during uninstallation")
-	f.BoolVar(&client.Purge, "purge", false, "remove the release from the store and make its name free for later use")
+	f.BoolVar(&client.KeepHistory, "keep-history", false, "remove all associated resources and mark the release as deleted, but retain the release history")
```

Comment:
Should the description of the flag be what setting the flag provides and not the whole command. So, something like the following:

"retains the release history including the release name"

Reply:
Everything in the description is what the flag provides. No other flag marks the release as deleted; they delete the release ledger entirely. I feel it's important to call that out so users aren't confused what the expected behaviour of this flag is supposed to be.

Reply:
I thought the `keep-history` flag retains the Helm history.  I would think the `helm uninstall` command remove all associated resources and mark the release as deleted.

To me, the flag is the differentiator.

Reply:
If I understand it right, with this change `helm uninstall` will delete the resources and remove all of the CRDs/secrets that contain the historical information. It's kind of like cleaning out the windows registry or app data when the app is uninstalled or removing all associated files from the Mac Library. With `--keep-history` those historical CRDs/secrets are kept around if you want to query them later.

Reply:
Yes but the flag imo should only describe what it does and that is keep the helm history.
Its a small point and I am ok with merging either way.